zebra-rpc/src/methods/get_block_template_rpcs/get_block_template.rs

```diff
@@ -253,11 +264,14 @@ where
             data: None,
         })?;
 
-    if let mempool::Response::FullTransactions(transactions) = response {
-        Ok(transactions)
-    } else {
+    let mempool::Response::FullTransactions {
+        transactions,
+        last_seen_tip_hash,
+    } = response else {
         unreachable!("unmatched response to a mempool::FullTransactions request")
-    }
+    };
+
+    Ok((last_seen_tip_hash == chain_tip_hash).then_some(transactions))
```

Comment:
Optional: or this comment might be enough

```suggestion
    // Check that the mempool and state were in sync when we made the requests
    Ok((last_seen_tip_hash == chain_tip_hash).then_some(transactions))
```